controllers/limitador_controller.go

```diff
@@ -199,6 +202,30 @@ func (r *LimitadorReconciler) reconcileDeployment(ctx context.Context, limitador
 		return err
 	}
 
+	// TODO: This block of code to be deleted when the upgrade path is no longer needed.
+	{
+		newDeployment := &appsv1.Deployment{}
+		if err := r.Client().Get(ctx, client.ObjectKeyFromObject(deployment), newDeployment); err != nil {
+			return err
+		}
+
+		availableCondition := helpers.FindDeploymentStatusCondition(newDeployment.Status.Conditions, "Available")
+
+		if availableCondition != nil && availableCondition.Status == v1.ConditionTrue {
+			oldDeployment := &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      limitadorObj.Name,
```

Comment:
I would create some function named `Release_v0_4_0_DeploymentName` in the package dedicated for upgrades

```go
func Release_v0_4_0_DeploymentName(limitadorObj *limitadorv1alpha1.Limitador) string {
	return limitadorObj.Name
}
```

Same for the configmap

Reply:
I got the functions for the v0.6.0 Deployment and ConfigMap names in the *upgrades* package. No underscores in the name though because of the complaining linter.